src/Std/Data/DHashMap/Internal/AssocList/Basic.lean

```diff
@@ -10,6 +10,11 @@ import Init.NotationExtra
 This is an internal implementation file of the hash map. Users of the hash map should not rely on
 the contents of this file.
 
+Note that some functions in this file (in particular, `foldrM`, `foldr`, `toList`, `replace`, and
+`erase`) are not tail-recursive. This is not a problem because they are only used internally by
+`HashMap`, where `AssocList` is always small. Before making this API public, we would need to add
+`@[csimp]` lemmas for tail-recursive implementations.
```

Comment:
If you prefer I can just add these now.

Reply:
Let's wait with this until we have a concrete plan to make `AssocList` public.